File: DevExpress-Demo/CS/GridDemo.Wpf/ModuleResources/DataAnnotations/DataAnnotationsElement2.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace GridDemo {
    public class DataAnnotationsElement2 {
        [Display(AutoGenerateField = false)]
        public int ID { get; set; }

        [Editable(false)]
        public string Employer { get; set; }

        [Display(Name = "First name"), Required]
        public string FirstName { get; set; }

        [Display(Name = "Last name"), Required]
        public string LastName { get; set; }

        [Display(Name = "Full name")]
        public string FullName { get { return FirstName + " " + LastName; } }

        [DisplayFormat(DataFormatString = "d2", ApplyFormatInEditMode = true)]
        public int Age { get; set; }

        [ReadOnlyAttribute(true)]
        public string SSN { get; set; }

        [DisplayFormat(NullDisplayText = "Department not set")]
        public string Department { get; set; }
    }
}

File: DevExpress-Demo/VB/GridDemo.Wpf/ModuleResources/DataAnnotations/DataAnnotationsElement2.vb
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations

Namespace GridDemo

    Public Class DataAnnotationsElement2

        <Display(AutoGenerateField:=False)>
        Public Property ID As Integer

        <Editable(False)>
        Public Property Employer As String

        <Display(Name:="First name"), Required>
        Public Property FirstName As String

        <Display(Name:="Last name"), Required>
        Public Property LastName As String

        <Display(Name:="Full name")>
        Public ReadOnly Property FullName As String
            Get
                Return FirstName & " " & LastName
            End Get
        End Property

        <DisplayFormat(DataFormatString:="d2", ApplyFormatInEditMode:=True)>
        Public Property Age As Integer

        <ReadOnlyAttribute(True)>
        Public Property SSN As String

        <DisplayFormat(NullDisplayText:="Department not set")>
        Public Property Department As String
    End Class
End Namespace
